Data_Engine/Modify/SimplifyRequestTree.cs

```diff
@@ -93,12 +93,17 @@ private static void Flatten(this IRequest request)
                                 subRequests.RemoveAt(i);
                             else if (subSub.Count == 1 && !(logical is LogicalNotRequest))
                             {
-                                subRequests.RemoveAt(i);
-
-                                if (subSub[0].GetType() == request.GetType())
-                                    subRequests.InsertRange(i, ((ILogicalRequest)subSub[0]).IRequests());
+                                if (request is LogicalNotRequest)
+                                    ((LogicalNotRequest)request).Request = subSub[0];
                                 else
-                                    subRequests.Insert(i, subSub[0]);
+                                {
+                                    subRequests.RemoveAt(i);
+
+                                    if (subSub[0].GetType() == request.GetType())
+                                        subRequests.InsertRange(i, ((ILogicalRequest)subSub[0]).IRequests());
+                                    else
+                                        subRequests.Insert(i, subSub[0]);
+                                }
```

Comment:
I'll be honest, this whole method is confusing me a lot. I get the purpose but I just don't get why it is so complex. Since refactoring this method is not the purpose of this PR, I'll ignore it but I cannot guarantee that this part of the code is bug-free. So I'll have to trust you on this one (with the assumption that this will be reworked soon anyways as discussed).

Reply:
Yeah, 100% agreed about everything above - happy to reiterate together with @alelom in the next milestone 👍